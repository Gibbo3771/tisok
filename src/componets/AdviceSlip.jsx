import React from "react";

const AdviceSlip = props => {
  return <div className="advice-text">{props.text}</div>;
};

export default AdviceSlip;
