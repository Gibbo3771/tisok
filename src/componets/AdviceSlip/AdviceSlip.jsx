// @flow
import React from "react";
import { Slip } from "../PropTypes";

export type Props = {
  slip: Slip
};

export default class AdviceSlip extends React.Component<Props> {
  render() {
    const { slip } = this.props;
    if (!slip.advice) return <div />;
    const animeProps = {
      opacity: [0, 1],
      delay: (element, index) => index * 15
    };
    return (
      <div id="advice-text" className="advice-text">
        {slip.advice.split("").map((v, i) => (
          <span key={i}>{v}</span>
        ))}
      </div>
    );
  }
}
