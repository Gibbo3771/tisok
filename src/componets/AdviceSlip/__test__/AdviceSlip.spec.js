// @flow
import React from "react";
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";
import { shallow } from "enzyme";
import AdviceSlip from "../AdviceSlip";
import sinon from "sinon";

Enzyme.configure({ adapter: new Adapter() });

// mock match type, this is used because sometimes id is indefined depending on use case
type Match = {
  params: {| id: number | void |}
};
// mock match
const match: Match = {
  params: { id: 1 }
};

const props = {
  onReady: sinon.spy(),
  match: match,
  slip: {
    slip_id: 1,
    advice: "always write tests first"
  }
};

describe("AdviceSlip", () => {
  beforeEach(() => {
    props.onReady.resetHistory();
  });
  it("renders correctly if it has a slip", () => {
    const { onReady, match, slip } = props;
    const tree = shallow(
      <AdviceSlip slip={slip} onReady={onReady} match={match} />
    );
    expect(tree).toMatchSnapshot();
  });
  it("renders correctly if it has an empty slip", () => {
    const { onReady, match } = props;
    const slip = {};
    const tree = shallow(
      <AdviceSlip onReady={onReady} match={match} slip={slip} />
    );
    expect(tree).toMatchSnapshot();
  });
  it("calls on ready with no id if no id is given", () => {
    const { onReady, match, slip } = props;
    // Undefined if route has no params
    // $ExpectError
    match.params.id = undefined;
    const tree = shallow(
      <AdviceSlip slip={slip} onReady={onReady} match={match} />
    );
    expect(onReady.calledOnceWith(undefined)).toBeTruthy();
  });
  it("calls on ready with the id if an id is given", () => {
    const { onReady, match, slip } = props;
    match.params.id = 1;
    const tree = shallow(
      <AdviceSlip slip={slip} onReady={onReady} match={match} />
    );
    expect(onReady.calledOnceWith(1)).toBeTruthy();
  });
});
