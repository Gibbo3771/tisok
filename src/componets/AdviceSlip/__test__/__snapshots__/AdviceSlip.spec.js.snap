// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdviceSlip renders correctly if it has a slip 1`] = `
<div
  className="advice-text"
  id="advice-text"
>
  <span
    key="0"
  >
    a
  </span>
  <span
    key="1"
  >
    l
  </span>
  <span
    key="2"
  >
    w
  </span>
  <span
    key="3"
  >
    a
  </span>
  <span
    key="4"
  >
    y
  </span>
  <span
    key="5"
  >
    s
  </span>
  <span
    key="6"
  >
     
  </span>
  <span
    key="7"
  >
    w
  </span>
  <span
    key="8"
  >
    r
  </span>
  <span
    key="9"
  >
    i
  </span>
  <span
    key="10"
  >
    t
  </span>
  <span
    key="11"
  >
    e
  </span>
  <span
    key="12"
  >
     
  </span>
  <span
    key="13"
  >
    t
  </span>
  <span
    key="14"
  >
    e
  </span>
  <span
    key="15"
  >
    s
  </span>
  <span
    key="16"
  >
    t
  </span>
  <span
    key="17"
  >
    s
  </span>
  <span
    key="18"
  >
     
  </span>
  <span
    key="19"
  >
    f
  </span>
  <span
    key="20"
  >
    i
  </span>
  <span
    key="21"
  >
    r
  </span>
  <span
    key="22"
  >
    s
  </span>
  <span
    key="23"
  >
    t
  </span>
</div>
`;

exports[`AdviceSlip renders correctly if it has an empty slip 1`] = `<div />`;
